### PR TITLE
[CHK-2042] feat(ecommerce-for-io): add PM mapping policy for endpoint `GET /wallets`

### PR DESCRIPTION
**Pull request by giovanniberti**
### List of changes

* add PM mapping policy for endpoint `GET /wallets`
* remove Wallet unused fields such as userId, contractId...

### Motivation and context

This PR is part of an effort to map the new eCommerce APIs to the current APIs exposed by Payment Manager (PM).

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+                byte[] bytes = new byte[16];
+                BitConverter.GetBytes((long)wallet["idWallet"]).CopyTo(bytes, 0);
+                result["walletId"] = new Guid(bytes).ToString();
+                string eCommerceWalletType = "";
+                string pmWalletType = (string) wallet["type"];
+                if (eCommerceWalletTypes.ContainsKey(pmWalletType)) {
+                    eCommerceWalletType = eCommerceWalletTypes[pmWalletType];
+                }
+                result["paymentMethodId"] = eCommercePaymentMethodIds[eCommerceWalletType];
+                result["status"] = "SCA_COMPLETED";
```
Comment (ciuffagianluca) [inline src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl:121]: Is this status name right?

File: src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+          <return-response>
+            <set-status code="502" reason="Bad Gateway" />
+        <set-body>
+            @{
+                JObject errorResponse = new JObject();
+                errorResponse["title"] = "Error starting session";
+                errorResponse["status"] = 502;
+                errorResponse["detail"] = "There was an error starting session for input wallet token";
+                return errorResponse.ToString();
+            }
```
Comment (giovanniberti) [inline src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl:20]:
```suggestion
            {
                "title": "Error starting session",
                "status": 502,
                "detail": "There was an error starting session for input wallet token"
            }
```

File: src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            <return-response>
+            <set-status code="502" reason="Bad Gateway" />
+            <set-body>
+                @{
+                    JObject errorResponse = new JObject();
+                    errorResponse["title"] = "Error retrieving user wallet data";
+                    errorResponse["status"] = 502;
+                    errorResponse["detail"] = "There was an error retrieving user wallet data";
+                    return errorResponse.ToString();
+                }
```
Comment (giovanniberti) [inline src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl:45]: Same here

File: src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+            <return-response>
+            <set-status code="404" reason="Wallet not found" />
+            <set-body>
+                @{
+                    JObject errorResponse = new JObject();
+                    errorResponse["title"] = "Wallet not found";
+                    errorResponse["status"] = 404;
+                    errorResponse["detail"] = "No wallet found for input wallet token";
+                    return errorResponse.ToString();
+                }
```
Comment (giovanniberti) [inline src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl:63]: Same here

File: src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+                    errorResponse["status"] = 502;
+                    errorResponse["detail"] = "There was an error retrieving user wallet data";
+                    return errorResponse.ToString();
+                }
+            </set-body>
+            </return-response>
+        </when>
+    </choose>
+    <set-variable name="pmUserWalletResponseBody" value="@(((IResponse)context.Variables["pmWalletResponse"]).Body.As<JObject>())" />
+    <set-variable name="pmUserWalletResponseBodyLength" value="@(((JArray)((JObject)context.Variables["pmUserWalletResponseBody"])["data"]).Count)" />
```
Comment (giovanniberti) [inline src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl:51]: Naming, what about

```suggestion
    <set-variable name="pmUserWalletsNumber" value="@(((JArray)((JObject)context.Variables["pmUserWalletResponseBody"])["data"]).Count)" />
```
?

File: src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+        <when condition="@(((IResponse)context.Variables["paymentMethodsResponse"]).StatusCode != 200)">
+            <return-response>
+            <set-status code="502" reason="Bad Gateway" />
+            <set-body>
+                @{
+                    JObject errorResponse = new JObject();
+                    errorResponse["title"] = "Error retrieving eCommerce payment methods";
+                    errorResponse["status"] = 502;
+                    errorResponse["detail"] = "There was an error retrieving eCommerce payment methods";
+                    return errorResponse.ToString();
```
Comment (giovanniberti) [inline src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl:85]: Same here

**Review by giovanniberti — COMMENTED** [review]
Just very minor comments

File: src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl (hunk 2 above)
Comment (pietro-tota) [reply src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl]: Fixed with https://github.com/pagopa/pagopa-infra/pull/1412/commits/cb63e0274613f0e985651bcafad6fbbd465e2986

File: src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl (hunk 6 above)
Comment (pietro-tota) [reply src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl]: Fixed with https://github.com/pagopa/pagopa-infra/pull/1412/commits/cb63e0274613f0e985651bcafad6fbbd465e2986

File: src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl (hunk 4 above)
Comment (pietro-tota) [reply src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl]: Fixed with https://github.com/pagopa/pagopa-infra/pull/1412/commits/cb63e0274613f0e985651bcafad6fbbd465e2986

File: src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl (hunk 5 above)
Comment (pietro-tota) [reply src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl]: I like it! Refactored with https://github.com/pagopa/pagopa-infra/pull/1412/commits/c36a278126a2107fa31cb76e7a834299992333bb

**Review by pietro-tota — DISMISSED** [review]

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**